tests/test_comparison_template_lib.py

```diff
@@ -0,0 +1,146 @@
+import pandas as pd
+import pytest
+
+import splink.duckdb.duckdb_comparison_library as cld
+import splink.duckdb.duckdb_comparison_template_library as ctld
+import splink.spark.spark_comparison_library as cls
+import splink.spark.spark_comparison_template_library as ctls
+
+from splink.duckdb.duckdb_linker import DuckDBLinker
+from splink.spark.spark_linker import SparkLinker
+
+
+def test_duckdb_date_comparison_run():
+    ctld.date_comparison("date")
+
+
+def test_spark_date_comparison_run():
+    ctls.date_comparison("date")
+
+
+def test_datediff_levels(spark):
+    df = pd.DataFrame(
+        [
+            {
+                "unique_id": 1,
+                "first_name": "Tom",
+                "dob": "2000-01-01",
+            },
+            {
+                "unique_id": 2,
+                "first_name": "Robin",
+                "dob": "2000-01-30",
+            },
+            {
+                "unique_id": 3,
+                "first_name": "Zoe",
+                "dob": "1995-09-30",
+            },
+            {
+                "unique_id": 4,
+                "first_name": "Sam",
+                "dob": "1966-07-30",
+            },
+            {
+                "unique_id": 5,
+                "first_name": "Andy",
+                "dob": "1996-03-25",
+            },
+            {
+                "unique_id": 6,
+                "first_name": "Alice",
+                "dob": "2000-03-25",
+            },
+            {
+                "unique_id": 7,
+                "first_name": "Afua",
+                "dob": "1960-01-01",
+            },
+        ]
+    )
+
+    # Generate our various settings objs
+    duck_settings = {
+        "link_type": "dedupe_only",
+        "comparisons": [ctld.date_comparison("dob")],
+    }
+
+    print(duck_settings)
+
+    spark_settings = {
+        "link_type": "dedupe_only",
+        "comparisons": [ctld.date_comparison("dob")],
```

Comment:
```suggestion
        "comparisons": [ctls.date_comparison("dob")],
```

Reply:
Ah, that is me being an idiot - thanks for spotting